pkg/vdr/key/creator.go

```diff
@@ -40,38 +45,42 @@ func (v *VDR) Create(keyManager kms.KeyManager, didDoc *did.Doc,
 		opt(createDIDOpts)
 	}
 
+	keyType, vmType := getKeyTypeAndVerMethodType(createDIDOpts.KeyType)
```

Comment:
This code is confusing.

If the user does not provide the key type option, it defaults to `Ed25519`, regardless of the verification method type they have set in `did.Doc`.

Reply:
if we don't need to create a **kms** key in `CreateDIDKey()` then this call is pointless

I will remove it and remove kms key creation as well

Reply:
after careful thought, I ended up not removing the kms key creation, to keep the original behaviour of creating a new key and a new *VerificationMethod if the didDoc has an empty list of []VerificationMethod